src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php

```diff
@@ -91,7 +91,9 @@ public function onAuthenticationFailure(Request $request, AuthenticationExceptio
 
         $this->logger?->debug('Authentication failure, redirect triggered.', ['failure_path' => $options['failure_path']]);
 
-        $request->getSession()->set(SecurityRequestAttributes::AUTHENTICATION_ERROR, $exception);
+        if ($request->hasPreviousSession()) {
```

Comment:
Better if we could check if `!$options['stateless']` here, but that option is not passed

Reply:
hasPreviousSession is absolutely not the same than stateless.

Reply:
So we need a way to access the `stateless` option from the firewall.
If someone has an idea how to do this, please help on that. I will close that PR for now.

Reply:
@Seb33300 You can check if the `_stateless` attribute is set to `true` instead

Reply:
Good point! I modified my PR and reopened it.

Reply:
Oups cannot reopen because I force pushed...
Here is the new PR: https://github.com/symfony/symfony/pull/51350